presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -679,6 +693,8 @@ private void alterTable(String databaseName, String tableName, Function<TableMet
 
         TableMetadata oldTableSchema = readSchemaFile("table", tableMetadataDirectory, tableCodec)
                 .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
+        checkVersion(oldTableSchema.getWriterVersion());
```

Comment:
maybe extract `readSechema`+`checkVersion` as helper method.

Reply:
i thought about that. I did not, because

- to avoid internal API inconcistency, i would need to create 5 writer and 5 reader methods, which would be a lot of boilerplate code
- one could still go directly to the existing method and would be still hard to catch this in review

thus i decided not to